front/tests/007-pathfinding.spec.ts

```diff
@@ -1,40 +1,53 @@
 import { test } from '@playwright/test';
 import { PlaywrightHomePage } from './pages/home-page-model';
+import { ProjectPage } from './pages/project-page-model';
+import { StudyPage } from './pages/study-page-model';
 import PlaywrightRollingstockModalPage from './pages/rollingstock-modal-model';
 import PlaywrightMap from './pages/map-model';
 import VARIABLES from './assets/operationStudies/test_variables';
 import PlaywrightScenarioPage from './pages/scenario-page-model';
 
 test.describe('Testing pathfinding', () => {
-  test('Test pathfinding: no electrified rolling stock for this path throws error', async ({
-    page,
-  }) => {
-    const playwrightHomePage = new PlaywrightHomePage(page);
-    const scenarioPage = new PlaywrightScenarioPage(page);
+  let playwrightHomePage: PlaywrightHomePage;
+  let projectPage: ProjectPage;
+  let studyPage: StudyPage;
+  let scenarioPage: PlaywrightScenarioPage;
+  let playwrightMap: PlaywrightMap;
+  let isProjectCreated = false;
+
+  test.beforeEach(async ({ page }) => {
+    playwrightHomePage = new PlaywrightHomePage(page);
+    projectPage = new ProjectPage(page);
+    studyPage = new StudyPage(page);
+    scenarioPage = new PlaywrightScenarioPage(page);
+    playwrightMap = new PlaywrightMap(page);
 
     await playwrightHomePage.goToHomePage();
 
-    // Real click on project, study, scenario
+    // Click on project, study
     await playwrightHomePage.goToOperationalStudiesPage();
-    await playwrightHomePage.page
-      .getByTestId('_@Test integration project')
-      .locator('div')
-      .getByRole('button')
-      .click();
-
-    await playwrightHomePage.page
-      .getByTestId('_@Test integration study')
-      .getByRole('button')
-      .click();
+    await projectPage.openProjectByTestId('_@Test integration project');
+    await studyPage.openStudyByTestId('_@Test integration study');
 
     // Create scenario with infra France
-    await scenarioPage.openScenarioCreationModal();
-    await scenarioPage.setScenarioName('_@Test integration scenario created local');
-    await scenarioPage.setSenarioInfraByName(VARIABLES.infraName);
-    const createButton = playwrightHomePage.page.getByText('Créer le scénario');
-    await createButton.click();
+    if (!isProjectCreated) {
```

Comment:
Maybe rename it with `isScenarioCreated` ?

Reply:
Done